File: App/Java/app/src/main/java/com/example/tonjies/templateproject/adapter/SelectAdapter.java
package com.example.tonjies.templateproject.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.tonjies.templateproject.R;

import java.util.List;

/**
 * Created by 舍长 on 2019/1/5
 * describe:
 */
public class SelectAdapter extends RecyclerView.Adapter<SelectAdapter.MyViewHolder> {

    private Context context;

    private List<String> mDates;

    public SelectAdapter(Context context, List<String> mDates) {
        this.context = context;
        this.mDates = mDates;
    }

    //填充View对象
    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = View.inflate(context, R.layout.item_city, null);
        return new MyViewHolder(view);
    }

    //加载每个item要是显示的数据
    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.textView.setText(mDates.get(position));
    }

    //返回item数量
    @Override
    public int getItemCount() {
        return mDates.size();
    }

    //初始化控件
    public class MyViewHolder extends RecyclerView.ViewHolder {
        private TextView textView;

        public MyViewHolder(View itemView) {
            super(itemView);
            textView = itemView.findViewById(R.id.tvCityName);
        }
    }
}

File: App/kotlins/app/src/main/java/com/example/tonjies/weatherx/adapter/SelectAdapter.kt
package com.example.tonjies.weatherx.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.example.tonjies.weatherx.R

class SelectAdapter : RecyclerView.Adapter<SelectAdapter.ViewHolder> {


    //上下文
    private var context: Context? = null
    //数据
    private var mList = ArrayList<String>()

    /**
     * 空参构造方法
     */
    constructor() {

    }

    constructor(context: Context, list: ArrayList<String>) {
        this.context = context
        this.mList = list
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_city, parent, false)
        return ViewHolder(view)
    }

    /**
     * 声明控件
     */
    class ViewHolder(item: View) : RecyclerView.ViewHolder(item) {
        //城市标题
        var city: TextView = item.findViewById(R.id.mTxtCityName)

    }

    override fun getItemCount(): Int {
        return mList.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val cityName = mList[position]
        holder.city.text = cityName
    }
}